inst/include/population_dynamics/population/population.hpp

```diff
@@ -33,15 +33,366 @@
+        
+        //std::vector<std::shared_ptr<fims::Fleet<Type>> > surveys;
+        Population() {
+            this->id = Population::id_g++;
+
+             
+        
+        }
+
+        //gets called when Initialize() function called in Information - just once at start of model run
+        void Initialize(int nyears, int nseasons, int nages) {
+          
+            
+            //size all the vectors to length of nages
+            nfleets = fleets.size();
+            ages.resize(nages);
+            catch_at_age.resize(nages);
+            catch_numbers_at_age.resize((nyears+1)*nages*nfleets);
+            mortality_F.resize(nyears*nages);
+            mortality_Z.resize(nages);
+            proportion_mature_at_age.resize((nyears+1)*nages);
+            initial_numbers.resize(nages);
+            weight_at_age.resize(nages);
+            catch_weight_at_age.resize(nyears*nages*nfleets);
+            unfished_numbers_at_age.resize((nyears+1)*nages);
+            numbers_at_age.resize((nyears+1)*nages);
+            expected_catch.resize(nyears*nfleets);
+            expected_index.resize(nyears*nfleets);
+            biomass.resize((nyears+1));
+            unfished_spawning_biomass.resize((nyears+1));
+            spawning_biomass.resize((nyears+1));
+            log_naa.resize(nages);
+            log_Fmort.resize(nfleets*nyears);
+            log_M.resize(nyears*nages);
+            log_q.resize(nfleets);
+            naa.resize(nages);
+            Fmort.resize(nfleets*nyears);
+            M.resize(nyears*nages);
+            q.resize(nfleets);
+
+            
+            
+        }
+
+        //gets called at each model iteration (used to zero out derived quantities, values summed with +=, etc.)
+        void Prepare() {
+          std::fill(unfished_spawning_biomass.begin(), unfished_spawning_biomass.end(), 0);
+          std::fill(spawning_biomass.begin(), spawning_biomass.end(), 0);
+          std::fill(mortality_F.begin(), mortality_F.end(), 0);
+          std::fill(expected_catch.begin(), expected_catch.end(), 0);
+
+          //Transformation Section
+          for (int age = 0; age <= this->nages; age++) {
+            this -> naa[age] = exp(this -> log_naa[age]);
+            for(int year = 0; year < this->nyears; year++) {
+              int index_ya = year*nages + age;
+              this -> M[index_ya] = exp(this -> log_M[index_ya]);
+            }
+          }
+
+          for(int fleet_ = 0; fleet_ <= this->nfleets; fleet_++) {
+            this -> Fmort[fleet_] = exp(this -> log_Fmort[fleet_]);
+            for(int year = 0; year < this->nyears; year++) {
+              int index_yf = year * nfleets + fleet_;
+              this -> q[index_yf] = exp(this -> log_q[index_yf]);
+            }
+          }
+          // call functions to set up recruitment deviations.
+          this -> recruitment -> PrepareConstrainedDeviations();
+          this -> recruitment.PrepareBiasAdjustment();
```

Comment:
->

Reply:
I think we put this in to test to try and understand the difference between -> and . to see if one of them failed to compile. We were going to ask you about the difference in one of the office hours.